File: prisma/refs/java/target-sum.java
import java.util.*;

class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        Map<Integer, Integer> counts = new HashMap<>();
        counts.put(0, 1);
        for (int n : nums) {
            Map<Integer, Integer> next = new HashMap<>();
            for (Map.Entry<Integer, Integer> e : counts.entrySet()) {
                next.merge(e.getKey() + n, e.getValue(), Integer::sum);
                next.merge(e.getKey() - n, e.getValue(), Integer::sum);
            }
            counts = next;
        }
        return counts.getOrDefault(target, 0);
    }
}

File: prisma/refs/py/target-sum.py
from collections import defaultdict


def find_target_sum_ways(nums, target):
    counts = {0: 1}
    for n in nums:
        nxt = defaultdict(int)
        for s, ways in counts.items():
            nxt[s + n] += ways
            nxt[s - n] += ways
        counts = nxt
    return counts.get(target, 0)
